### PR TITLE
Add simple deployment to emit podstatus to riemann every minute

### PR DESCRIPTION
**Pull request by cnelson**
In order to complete  https://favro.com/card/1e11108a2da81e3bd7153a7a/18F-795

### CODE REVIEW

File: manifests/riemann-podstatus-template.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  name: riemann-podstatus
+  labels:
+    app: riemann-podstatus
+spec:
+  template:
+    metadata:
+      labels:
+        name: riemann-podstatus
+    spec:
+      hostNetwork: true
```
Comment (jmcarp) [inline manifests/riemann-podstatus-template.yaml]: Remind me why we need this?

File: manifests/riemann-podstatus-template.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        args:
+        - "-c"
+        - |-
+          while sleep 60; do
+            /bin/bash -x <(curl -s -k -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_PORT_443_TCP_PORT/api/v1/pods/ | jq -r '.items[] | "riemannc send --server '${RIEMANN_HOST}' --host riemann-podstatus --service "+.metadata.namespace+"/"+.metadata.name+" --state "+.status.phase+";"')
+          done
+        env:
+        - name: RIEMANN_HOST
+          value: (( grab meta.riemann ))
+        securityContext:
```
Comment (jmcarp) [inline manifests/riemann-podstatus-template.yaml]: Same question as `hostNetwork`.

**Comment by jmcarp** [comment]
I'm wondering why this runs as a `DaemonSet`--it looks like each node running the script should be sending the same results to riemann.

File: manifests/riemann-podstatus-template.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      containers:
+      - image: 18fgsa/concourse-task
+        name: bash
+        command:
+        - "/bin/bash"
+        args:
+        - "-c"
+        - |-
+          while sleep 60; do
+            /bin/bash -x <(curl -s -k -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" https://$KUBERNETES_SERVICE_HOST:$KUBERNETES_PORT_443_TCP_PORT/api/v1/pods/ | jq -r '.items[] | "riemannc send --server '${RIEMANN_HOST}' --host riemann-podstatus --service "+.metadata.namespace+"/"+.metadata.name+" --state "+.status.phase+";"')
```
Comment (jmcarp) [inline manifests/riemann-podstatus-template.yaml]: IMO this would be easier to read with some linebreaks.

**Comment by cnelson** [comment]
> I'm wondering why this runs as a DaemonSet--it looks like each node running the script should be sending the same results to riemann.

For redundancy; it doesn't hurt to have these metrics come in more than once but what alternate approach would you suggest?

**Comment by jmcarp** [comment]
I don't think there's anything wrong with using a DaemonSet, just want to make sure I understand what's happening. I would think we could also use a Deployment or ReplicaSet with >1 replicas, but that might be more likely to fail if all the replicas get scheduled to the same node and something happens to it.

**Pull request merged**